Database docs

olijeffers0n opened the pull request:
Initial Work for database docs. I made these on a plane so some sections may be lacking. Let me know of any additions needed on the PR or discord :)

closed #106

MiniDigger commented on docs/paper/dev/misc/databases.md:
```diff
@@ -0,0 +1,107 @@
+
+## MySQL vs SQLite
+
+There are two main types of database software that you could use: MySQL vs SQLite. They both have many pros and cons outlined [here](https://www.hostinger.com/tutorials/sqlite-vs-mysql-whats-the-difference/)
```
this is meh, instead of linking products, we should talk about file based dbs (like sqlite, h2, whatever) vs standalone dbs (postgres, mysql, etc).
that article in particular is worthless seo spam

MiniDigger commented on docs/paper/dev/misc/databases.md:
```diff
@@ -0,0 +1,107 @@
+
+### Usage
+
+From here, you must invoke a `Class.forName` on the driver to allow it to initialise and then create the connection to the database:
```
is this still the case on modern systems? I thought it does stuff automatically nowadays

olijeffers0n replied:
Yes it is needed, Spigot ships an archaic version of JDBC (3.42.0.0) which does not support the automatic init - (V4+)

MiniDigger commented on docs/paper/dev/misc/databases.md:
````diff
@@ -0,0 +1,107 @@
+}
+```
+
+To learn more about the Java Database Connection, see [here](https://www.w3schools.blog/jdbc-tutorial)
````
w3schools is a really meh resource (and that link isnt even the real one?)
I would suggest this as a quick primer https://www.baeldung.com/java-jdbc and https://docs.oracle.com/javase/tutorial/jdbc/basics/index.html for a fully featured reference

MiniDigger commented on docs/paper/dev/misc/databases.md:
```diff
@@ -0,0 +1,107 @@
+## MySQL
+
+Working with MySQL requires a few more steps, however it can offer performance benefits for larger databases with 
+many tables and concurrent accesses. We will be focussing on the [Hikari](https://github.com/brettwooldridge/HikariCP) library for this tutorial. First, add the
```
we should have a sentence or two about what a connection pool is and why its relevant for standalone dbs like postgres and mysql

powercasgamer commented:
Should maybe also mention that for gradle you need to use shadow, or the library loader?

olijeffers0n commented:
With response to Mini's comments - A more general overview of what you should be researching could be useful. I'll probably focus more on this angle in future

olijeffers0n commented:
@MiniDigger I have addressed your comments I think. I kept the basic guides within dropdowns, and added a more general overview. Would be happy for another review when you get time 😄

A248 commented on docs/paper/dev/misc/databases.mdx:
```diff
@@ -0,0 +1,208 @@
+## SQLite
+
+To work with SQLite you will need a driver to connect / initialise the database. 
+We will be recommending the JDBC connector which can be added to your project with the following dependency:
```
The dependency declaration isn't useful because calling classes in `org.sqlite` packages is unnecessary. No dependency is needed; the whole interface works seamlessly with JDBC.

olijeffers0n replied:
Good point actually, this is only needed if you intend to relocate a different version of JDBC

kennytv commented on docs/paper/dev/misc/databases.mdx:
```diff
@@ -0,0 +1,191 @@
+NoSQL databases use various data models, such as key-value, document, column-family, or graph, depending on 
+the specific requirements of the application.
+
+On the other hand, An SQL database is a type of database management system that follows the relational database model. 
```
```suggestion
On the other hand, an SQL database is a type of database management system that follows the relational database model. 
```

kennytv commented on docs/paper/dev/misc/databases.mdx:
```diff
@@ -0,0 +1,191 @@
+
+## SQLite
+
+To work with SQLite you will need a driver to connect / initialise the database. 
```
```suggestion
To work with SQLite, you will need a driver to connect / initialize the database. 
```

olijeffers0n commented:
Addressed Comments
